src/backends/quadiron/quadiron_fnt_common.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2018 Scality
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.  THIS SOFTWARE IS PROVIDED BY
+ * THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * quadiron_fnt backend implementation
+ *
+ * vi: set noai tw=79 ts=4 sw=4:
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "erasurecode.h"
+#include "erasurecode_backend.h"
+#include "erasurecode_helpers.h"
+#include "erasurecode_helpers_ext.h"
+#include "quadiron_fnt_common.h"
+
+typedef void (*quadiron_hex_dump_func)(uint8_t *buf, size_t size);
+quadiron_hex_dump_func quadiron_hex_dump;
+
+int quadiron_fnt_encode(void *desc, char **data, char **parity,
+		int blocksize)
+{
+	quadiron_fnt_descriptor *qdesc = (quadiron_fnt_descriptor*) desc;
+	int qwanted_idxs[qdesc->k + qdesc->m];
+	unsigned int i;
+
+	for (i = 0; i < qdesc->k + qdesc->m; i++) {
```

Comment:
could use `memset` here, no?

Reply:
agree but mostly copy-paste from other drivers

Reply:
I don't think that's a good justification :slightly_smiling_face: